vm.go

```diff
@@ -327,7 +327,7 @@ func (vm *vm) setFuncName(s unistring.String) {
 
 func intToValue(i int64) Value {
 	if i >= -maxInt && i <= maxInt {
-		if i >= -128 && i <= 127 {
+		if i >= -128 && i <= (intCacheSize-129) {
```

Comment:
Adding my math since I had to wrap my head around this..

If we assume we set intCacheSize to 256. Then we would still get the correct existing figure (256-129=127)

Reply:
[opt/q] sorry, my bad for approving too early :( 
Do you think we should perform the subtraction (`intCacheSize-129`) within this function? It gets called fairly frequently so it might be worth moving that out.  You could move it into a const. Here's an example:
```golang
const  (
  intCacheSize = 16384
  intCacheMinValue = -128
  intCacheMaxValue = intCacheMinValue + intCacheSize - 1
```

```suggestion
		if i >= intCacheMinValue && i <= intCacheMaxValue {
```
It might also be worth adding a very quick test to confirm that the calculations are correct and we won't panic when converting values that go from `intCacheMinValue - 1` until `intCacheMaxValue + 1`.

Reply:
Ah good catch! My bad for jumping the gun on the merge 😅